x-pack/plugins/enterprise_search/public/applications/shared/result/result_title.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SearchHit } from '@elastic/elasticsearch/lib/api/types';
+
+const TITLE_KEYS = ['title', 'name'];
+
+const isRecord = (source: unknown | undefined): source is Record<PropertyKey, unknown> => {
+  return source !== undefined && typeof source === 'object' && source !== null;
```

Comment:
question: Wouldn't just `typeof source === 'object' && source !== null` not already do the trick here? An object is never undefined, right?

Reply:
I think you're correct. I'll double check.